project-rome-docs/objectivec-api/remotesystems.commanding/MCDRemoteSystemAppRegistration.md

```diff
@@ -10,60 +10,88 @@ keywords: microsoft, windows, iOS, iPhone, objectiveC, connected devices, Projec
 @interface MCDRemoteSystemAppRegistration : NSObject
 ```  
 
-This class represents an application that is to be registered with the Connected Devices platform.
-Registration is required (**saveAsync** must have successfully completed) to receive responses from commands as well as to allow other apps to discover this application and its attributes and capabilities.
+This class contains all of the information about this app that another could discover and use.
+
+> [!NOTE] 
+> MCDRemoteSystemAppRegistration information must be published before any outgoing communication to another app is possble. This is so that the other application can know how to respond to that communication.
 
 ## Properties
 
 ### account
 `@property(nonatomic, readonly, nullable) MCDConnectedDevicesAccount* account;`
 
-The current MCDConnectedDevicesAccount account provided.
+Account that this registration belongs to.
 
 ### attributes
 `@property(nonatomic, copy, nullable) NSDictionary<NSString*, NSString*>* attributes;`
 
-A **Map** of attributes to set in the registration.
+ Dictionary of strings that describe the attributes of this app.
 
 ### appServiceProviders
 `@property(nonatomic, copy, nullable) NSArray<id<MCDAppServiceProvider>>* appServiceProviders;`
 
-The app service providers associated with the registration. New values are made available to discovering apps and devices via the **saveAsync** call.
+Array of AppServiceProviders that this app supports.
+
+> [!NOTE] 
+> An app service provider must be present in (though not necessarily published) this array in order to receive incoming connections.
 
 ### launchUriProvider
 `@property(nonatomic, readwrite, nullable) id<MCDLaunchUriProvider> launchUriProvider;`
 
-The launch uri provider associated with this application.
+Launch Uri provider for this app.
+
+> [!NOTE] 
+> A launch uri provider must be stored in (though not necessarily published) this property in order to receive incoming requests.
```

Comment:
I'm having trouble parsing the "(though not necessarily published)" clause here. Should it be "(though not necessarily published **to** )"? Does it belong elsewhere in this sentence?

Reply:
Added some clarity to help.